File: JavaScript/String/L_uncommonWordInStr.js
var uncommonFromSentences = function (s1, s2) {
    let set = new Set();
    let dupset = new Set();
    s1 = s1.split(" ");
    s2 = s2.split(" ");
    for (let x of s1) {
        if (!dupset.has(x)) {
            if (set.has(x)) {
                set.delete(x);
                dupset.add(x);
            } else set.add(x);
        }
    }
    for (let x of s2) {
        if (!dupset.has(x)) {
            if (set.has(x)) {
                set.delete(x);
                dupset.add(x);
            } else set.add(x);
        }
    }
    return Array.from(set);
};

File: Python/Strings/L_uncommonWordInStr.py
def uncommonFromSentences(s1: str, s2: str):
        s1_count = {}
        s2_count = {}
        res = []
        for word in s1.split(" "):
            s1_count[word] = s1_count.get(word, 0) + 1
        for word in s2.split(" "):
            s2_count[word] = s2_count.get(word, 0) + 1
        for key, val in s1_count.items():
            if val == 1 and key not in s2_count:
                res.append(key)
        for key, val in s2_count.items():
            if val == 1 and key not in s1_count:
                res.append(key)
        return res
